torchci/pages/userbenchmark_view.tsx

```diff
@@ -0,0 +1,156 @@
+import { Grid } from "@mui/material";
+import TimeSeriesPanel from "components/metrics/panels/TimeSeriesPanel";
+import dayjs from "dayjs";
+import { RocksetParam } from "lib/rockset";
+import React, { useState } from "react";
+import styles from "components/hud.module.css";
+
+const ROW_HEIGHT = 240;
+type UserbenchmarkProps = {};
+type UserbenchmarkState = {url: string | null, content: any};
+class UserbenchmarkResults extends React.Component<UserbenchmarkProps, UserbenchmarkState> {
+    constructor(props: UserbenchmarkProps) {
+        super(props);
+
+        this.state = {
+            url: null,
+            content: null,
+        };
+    }
+
+    // returns a list[list[data]]
+    // outer list: first a row of headers; and then rows of data.
+    // data order in the inner list:
+    //   0. Metric name
+    //   1. Base revision latency
+    //   2. PR revision latency
+    //   3. Speedup from the PR revision
+    parseCsv(msg: string) {
+        let as_array = msg.split("\n").map((x: any) => x.split(";"));
+        as_array = as_array.filter((subarray: any[]) => subarray.length == 3);
+        as_array[0].push("Speedup");
+        for (let i=1; i<as_array.length; ++i) {
+            for (let j=1; j<=2; ++j) {
+                as_array[i][j] = parseFloat(as_array[i][j]);
+            }
+            const base_value = as_array[i][1] as number;
+            const pr_value = as_array[i][2] as number;
+            as_array[i].push(base_value / pr_value - 1);
+        }
+        return as_array;
+    }
+
+    getHeader(row: any[]) {
+        const metric_header = row[0] as string;
+        const base_revision = row[1] + " (base)" as string;
+        const change_revision = row[2] + " (change)" as string;
+        const speedup_header = row[3] as string;
```

Comment:
This seems like a good candidate for either a class or a [Named Tuple](https://www.geeksforgeeks.org/typing-namedtuple-improved-namedtuples/).  Getting the data indexes right can get tricky

Reply:
Added a class for this. But we still have to rely on indices when we first load the data from the csv.